apps/admin-server/src/pages/projects/[project]/widgets/resourcesmap/[id]/buttons.tsx

```diff
@@ -0,0 +1,220 @@
+import { useState } from 'react';
+import { Button } from '@/components/ui/button';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { Separator } from '@/components/ui/separator';
+import { Heading } from '@/components/ui/typography';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { useFieldDebounce } from '@/hooks/useFieldDebounce';
+import type {ResourceOverviewMapWidgetProps} from '@openstad-headless/leaflet-map/src/types/resource-overview-map-widget-props'
+import { EditFieldProps } from '@/lib/form-widget-helpers/EditFieldProps';
+import * as z from 'zod';
+
+const formSchema = z.object({
+  ctaButton: z.object({
+    show: z.boolean(),
+    label: z.string().optional(),
+    href: z.string().optional(),
+  }),
+  countButton: z.object({
+    show: z.boolean(),
+    label: z.string().optional(),
+  }),
+});
+
+export default function WidgetResourcesMapButton(
+  props: ResourceOverviewMapWidgetProps &
+    EditFieldProps<ResourceOverviewMapWidgetProps>
+) {
+
+  type FormData = z.infer<typeof formSchema>;
+
+  async function onSubmit(values: FormData) {
+    props.updateConfig({ ...props, ...values });
+  }
+
+  const { onFieldChange } = useFieldDebounce(props.onFieldChanged);
+
+  const form = useForm<FormData>({
+    resolver: zodResolver<any>(formSchema),
+    defaultValues: {
+      countButton: props?.countButton,
+      ctaButton: props?.ctaButton,
+    },
+  });
+
+  const [showCtaFields, setShowCtaFields] = useState(props?.ctaButton?.show || false)
+  const [showCountFields, setShowCountFields] = useState(props?.countButton?.show || false)
+
+  return (
+    <div className="p-6 bg-white rounded-md">
+      <Form {...form}>
+        <Separator className="my-4" />
+        <form
+          onSubmit={form.handleSubmit(onSubmit)}
+          className="space-y-4 lg:w-1/2">
+
+          <Heading size="xl">Call-To-Action knop</Heading>
+          <FormField
+            control={form.control}
+            name="ctaButton.show"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>
+                Toon een &apos;Call To Action&apos; knop
+                </FormLabel>
+                <Select
+                  onValueChange={(value) => {
+                      let val = value == 'true' ? true : false
+                      onFieldChange(field.name, val);
```

Comment:
Ik zou de debounce onFieldChange niet gebruiken in een select,  maar direct de props.onFieldChanged aanroepen. De debounce hook wil je eigenlijk alleen doen voor text input fields, zodat niet bij elke keypress er een preview load call wordt gedaan. 

Bij een select change in een <Select> control wil je dit juist wel in 1 keer zonder een 300ms vertraging

Reply:
Ah, is dat het idee :)

Ik heb het aangepast, maar weet dat die debounce niet lekker werkt, want ik krijg wel degelijk updates van de preview bij elke toets aanslag. Ik zet het in de excel.